### PR TITLE
[FIX] web: nested x2many context fix

### PR DESCRIPTION
**Pull request by brboi**
# Before this commit
Suppose that in a form view, there is an x2many list view that would open a form view dialog when edited.
For the dialog to open, the concerned record is duplicated and passed in props to a FormRenderer.
Now suppose that this form view dialog also has an x2many list view rendered somewhere: the StaticList datapoint attached to this list will have the same context as its parent. This is not what we want because some of the keys of the context does not make sense with the list's model.
In fact, it has led to tracebacks to be displayed in case of we want to open a record of this list in another form view dialog.

# After this commit
When those lists are created, their context are cleared from their parents' keys starting with "default_" or ending with "_view_ref".

### CODE REVIEW

**Comment by kebeclibre** [comment]
in legacy we wiped out _view_refs at form_view.js:L124 with 
``` 
                    // Remove *_view_ref coming from parent view
                    var refinedContext = _.pick(self.loadParams.context, function (value, key) {
                        return key.indexOf('_view_ref') === -1;
                    });
```
we did that once more in view_dialogs.js:L202; when displaying a x2m record in a modal

for the default_ keys, we wiped them out each time we created a record inside a X2M list basicModel: _processX2ManyCommands and _addX2ManyDefaultRecord


In the basicRelational model, I think what you did is not wrong. But I think wiping out _view_ref should happen before the root record is created ; and default_values in the setup of staticList.
(and direclty in the internal of __bm__)

Those are complicated to get around, so I propose you test this and if it doesn't work go back to your solution

**Comment by kebeclibre** [comment]
also, make sure that x2m field having a context in which _view refs and default_ are defined

File: addons/web/static/tests/views/form/form_view_tests.js [hunk 1]
```diff
@@ -942,6 +942,93 @@ QUnit.module("Views", (hooks) => {
...
+                    }"/>
+                </form>`,
+            resId: 2,
+            mockRPC: (route, { method, model, kwargs }) => {
+                if (method === "get_views") {
+                    const { context } = kwargs;
+                    assert.step(model);
+                    assert.deepEqual(context, expectedContexts[model]);
+                    delete expectedContexts[model];
+                }
```
Comment (kebeclibre) [inline addons/web/static/tests/views/form/form_view_tests.js:999]: needs to assert the first onchange (with a new record -- in a new test ?)

**Review by aab-odoo — APPROVED** [review]
wow, there are still bugs in the basic model, and this is not a small one, I'm a bit surprised... Anyway the fix looks good.

robodoo r+

**Pull request closed without merging**